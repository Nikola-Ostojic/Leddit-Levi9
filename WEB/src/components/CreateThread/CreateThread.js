import React, { useState } from "react";
import { Button, TextField } from "@material-ui/core";
import { toast } from "react-toastify";

import { withRouter } from "react-router-dom";

import { apiCall } from "utils/ApiCall";

const CreateThread = ({ history }) => {
  const [title, setTitle] = useState("");
  const [content, setContent] = useState("");

  const createThread = (e) => {
    e.preventDefault();
    apiCall({
      url: "/threads",
      method: "POST",
      data: { title, content },
      requiresAuth: true
    })
      .then(() => {
        toast.success(`Thread: ${title} created!`);
        // Cleaning up the state after a successful creation
        setTitle("");
        setContent("");
      })
      .catch((e) => {
        toast.error("Failed to create thread");
      })
      .finally(() => history.goBack());
  };

  return (
    <form onSubmit={createThread}>
      <TextField
        fullWidth
        margin="normal"
        label="Thread name"
        type="text"
        required
        onChange={({ target: { value } }) => setTitle(value)}
        value={title}
      />
      <TextField
        fullWidth
        required
        margin="normal"
        label="Content"
        type="text"
        onChange={({ target: { value } }) => setContent(value)}
        value={content}
      />
      <Button variant="contained" color="primary" type="submit">
        Create thread
      </Button>
    </form>
  );
};

export default withRouter(CreateThread);
