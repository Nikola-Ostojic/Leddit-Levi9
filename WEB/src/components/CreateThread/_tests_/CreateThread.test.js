import React from "react";
import { shallow } from "enzyme";

import CreateThread from "../CreateThread";

describe("CreateThread component tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<CreateThread />);

    expect(wrapper).toMatchSnapshot();
  });
});
