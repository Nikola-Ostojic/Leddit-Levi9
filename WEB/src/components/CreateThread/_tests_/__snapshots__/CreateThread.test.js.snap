// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateThread component tests Should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
